build.go

```diff
@@ -125,6 +126,7 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 		HTTPProxy:  proxyConfig.HTTPProxy,
 		HTTPSProxy: proxyConfig.HTTPSProxy,
 		NoProxy:    proxyConfig.NoProxy,
+		Network:    opts.Network,
```

Comment:
I am not sure this belongs in `ProxyConfig`. Can we add a `ContainerConfig` field to `LifecycleOptions` with a `Network` field. This would give us a nice way to extend to include future `docker run` type flags